### PR TITLE
Add new endpoint `/_api/key-generators`

### PR DESCRIPTION
**Pull request by nerpaula**
### Description

See https://github.com/arangodb/arangodb/pull/20374

#### Upstream PRs

- 3.10:
- 3.11:
- 3.12: arangodb/enterprise-preview:devel-nightly

### CODE REVIEW

File: site/content/3.12/develop/http-api/collections.md [hunk 1]
```diff
@@ -950,6 +950,45 @@ logJsonResponse(response);
...
+        on success:
+
+        - `padded`
+        - `uuid`
+        - `autoincrement`
+        - `traditional`
+      responses:
+        '200':
+          description: |
+            The key generators that were found.
```
Comment (Simran-B) [inline site/content/3.12/develop/http-api/collections.md:973]: I believe this is the correct way to describe the response. We should check what Swagger UI displays (should be a multi-select widget)

```suggestion
            The available generators for document keys.
          content:
            application/json:
              schema:
                type: array
                uniqueItems: true
                items:
                  type: string
                  enum:
                    - padded
                    - uuid
                    - autoincrement
                    - traditional
```

File: site/content/3.12/release-notes/version-3.12/api-changes-in-3-12.md [hunk 2]
```diff
@@ -111,7 +111,15 @@ The default value remains 128 MiB.
 
 ### Endpoints added
 
+#### Collections API
 
+You can now retrieve the key generators for collections using the following
+endpoint:
+
+`GET /_api/key-generators`
```
Comment (Simran-B) [inline site/content/3.12/release-notes/version-3.12/api-changes-in-3-12.md:119]:
```suggestion
You can now retrieve the available key generators for collections using the new
`GET /_api/key-generators` endpoint.
```

File: site/content/3.12/develop/http-api/collections.md [hunk 3]
```diff
@@ -950,6 +950,45 @@ logJsonResponse(response);
...
+
+**Examples:**
+
+```curl
+---
+description: |-
+  Retrieving the key generators of collections:
+name: RestCollectionGetKeyGenerators
+---
+var url = "/_api/key-generators/"
```
Comment (Simran-B) [inline site/content/3.12/develop/http-api/collections.md:986]:
```suggestion
var url = "/_api/key-generators"
```

File: site/content/3.12/release-notes/version-3.12/api-changes-in-3-12.md [hunk 4]
```diff
@@ -111,7 +111,15 @@ The default value remains 128 MiB.
...
+
+See the [description](../../develop/http-api/collections.md#get-the-available-key-generators)
```
Comment (Simran-B) [inline site/content/3.12/release-notes/version-3.12/api-changes-in-3-12.md:121]:
```suggestion
See the [HTTP API description](../../develop/http-api/collections.md#get-the-available-key-generators)
```

**Review by Simran-B — CHANGES_REQUESTED** [review]

File: site/content/3.12/develop/http-api/collections.md [hunk 5]
```diff
@@ -950,6 +950,57 @@ logJsonResponse(response);
...
+
+        The call returns a JSON object with the available key generators
+        on success:
+
+        - `padded`
+        - `uuid`
+        - `autoincrement`
+        - `traditional`
+      responses:
+        '200':
```
Comment (nerpaula) [inline site/content/3.12/develop/http-api/collections.md:971]: @Simran-B I applied your suggestion but the output is not right, see https://deploy-preview-420--docs-hugo.netlify.app/devel/develop/http-api/collections/#get-the-available-key-generators_res_200

Comment (Simran-B) [reply site/content/3.12/develop/http-api/collections.md:971]: Seems fine in Swagger UI and swagger-cli validated it as correct, so it looks like the toolchain is not able to render a construct like this

image

File: site/content/3.12/develop/http-api/collections.md [hunk 6]
```diff
@@ -950,6 +950,45 @@ logJsonResponse(response);
...
+          description: |
+            A list of the available generators for document keys:
+            - `"padded"`
+            - `"uuid"`
+            - `"autoincrement"`
+            - `"traditional"`
+          content:
+            application/json:
+              schema:
+                type: array
```
Comment (Simran-B) [inline site/content/3.12/develop/http-api/collections.md:973]: This is actually incorrect (my bad), it's an object at the top-level, and an array under the `keyGenerators` property. The toolchain basically doesn't seem to support arrays, which is a problem here but also for a few other endpoints (and more in the future if we move from full-text to structural descriptions)

File: site/content/3.12/develop/http-api/collections.md (hunk 5 above)
Comment (Simran-B) [reply site/content/3.12/develop/http-api/collections.md:963]: With added example value:
image

**Review by Simran-B — APPROVED** [review]

File: site/content/3.12/develop/http-api/collections.md (hunk 5 above)
Comment (nerpaula) [reply site/content/3.12/develop/http-api/collections.md:963]: Looks great!

**Pull request merged**